crates/cargo-wdk/src/actions/build.rs

```diff
@@ -30,7 +33,8 @@ pub enum BuildActionError {
 /// Action that orchestrates building of driver project using cargo command.
 pub struct BuildAction<'a> {
     package_name: &'a str,
-    profile: &'a Profile,
+    profile: Option<Profile>,
+    target_arch: Option<CpuArchitecture>,
```

Comment:
Its efficient use Option<&'a CpuArchitecture> here.

Reply:
When I think about this again, the use of the word "efficient" is not applicable in this case since `CpuArchitecture` enum type takes same size (due to alignment) on stack as when it's passed as reference. So, the current and proposed approaches will have same cost.

nit: Now the cost concerns out of the way, to be consistent with other params we can make them `Option<&'a Profile>` & `Option<&'a CpuArchitecture>`. But this is just a nit now.

Reply:
A reference will actually have a a greater cost. With the value you pay for the copy and with a reference you pay for copy and a dereference.